typescript/cli/src/commands/deploy.ts

```diff
@@ -67,48 +67,79 @@ const agentCommand: CommandModuleWithContext<{
 };
 
 /**
- * Core command
+ * Generates a command module for deploying Hyperlane contracts.
+ *
+ * @param deployFunction - A function that performs the actual deployment
+ * @returns A command module used to deploy Hyperlane contracts.
  */
-const coreCommand: CommandModuleWithWriteContext<{
-  targets: string;
-  ism?: string;
-  hook?: string;
-  'dry-run': string;
-  'from-address': string;
-  agent: string;
-}> = {
-  command: 'core',
-  describe: 'Deploy core Hyperlane contracts',
+export const deployWith = (
+  deployFunction: (params: {
+    context: WriteCommandContext;
+    chain: ChainName;
+    configFilePath: string;
+  }) => Promise<HyperlaneAddresses<any>>,
+): CommandModuleWithWriteContext<{
+  config: string;
+  chain: string;
+  dryRun: string;
+}> => ({
+  command: CRUD_COMMANDS.DEPLOY,
+  describe: 'Deploy Hyperlane contracts',
   builder: {
-    targets: coreTargetsCommandOption,
-    ism: ismCommandOption,
-    hook: hookCommandOption,
-    agent: agentConfigCommandOption(false, './configs/agent.json'),
+    config: {
+      type: 'string',
+      description: 'The path to a JSON or YAML file with a deployment config.',
+      demandOption: true,
+    },
+    chain: {
+      type: 'string',
+      description: 'The name of a single chain to deploy to',
+    },
     'dry-run': dryRunCommandOption,
-    'from-address': fromAddressCommandOption,
   },
-  handler: async ({ context, targets, ism, hook, agent, dryRun }) => {
-    logGray(
-      `Hyperlane permissionless core deployment${dryRun ? ' dry-run' : ''}`,
-    );
+  handler: async ({ context, chain, config: configFilePath, dryRun }) => {
+    const { chainMetadata, isDryRun, dryRunChain, registry, skipConfirmation } =
+      context;
+
+    logGray(`Hyperlane permissionless deployment${dryRun ? ' dry-run' : ''}`);
     logGray(`------------------------------------------------`);
 
+    // Select a dry-run chain if it's not supplied
+    if (dryRunChain) {
+      chain = dryRunChain;
+    } else if (!chain) {
+      if (skipConfirmation) throw new Error('No chain provided');
+      chain = await runSingleChainSelectionStep(
+        chainMetadata,
+        'Select chain to connect:',
+      );
+    }
```

Comment:
It's not a big deal but I think it would be a good idea to keep these yaml command handlers short and simple. Put the logic for running through the various steps somewhere else like we did before. The benefit of this is that it separates concerns. command line arg parsing -> Yargs, deployment -> something else

Reply:
Removed extra logic into deployFunctions